pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -192,18 +165,10 @@ func (c *controller) createInformer(
 		cache.ResourceEventHandlerFuncs{
 			// TODO: filtering functions to skip over un-referenced resources (perf)
 			AddFunc: func(obj interface{}) {
-				if err := vf(obj); err != nil {
-					handleValidationFailure(obj, err)
-					return
-				}
 				incrementEvent(otype, "add")
 				c.queue.Push(kube.NewTask(handler.Apply, nil, obj, model.EventAdd))
 			},
 			UpdateFunc: func(old, cur interface{}) {
-				if err := vf(cur); err != nil {
```

Comment:
If we remove validation from here, the invalid objects will go to Event Handlers as an example ServiceDiscovery's event handler here https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/external/servicediscovery.go#L63  which tries to convert it to services and that might create issues like panic etc?

Reply:
I think we should guard this and addFunc with the same flag that I introduced and keep the validation as is.

Reply:
@ramaraochavali Almost forget the panic before your fix. The root cause here is we can not make sure if a field exist. But this is not the responsibility of callback, the callback should just convert object. And this procedure is safely verified for a long time.

```
			for _, w := range d.Route {
				w.Destination.Host = string(ResolveShortnameToFQDN(w.Destination.Host, r.ConfigMeta))
			}
```

BTW, i will file a pr adding a check.

Reply:
Thanks for adding that PR, that looks good.

>  the callback should just convert object

Are you sure conversion does not cause any panic if there are missing fields like the virtual case?  or are you saying it is the responsibility of the conversion logic to ensure it does not run in to that situation?

Reply:
Yes, basically the conversion is to convert a IstioObject to `model.Config`. Which is not aware of specific API. As you listed for ServiceEntry it is an exception, it will convert `model.Config` to `model.Service`.  Whatever it is just a conversion, it should be responsible to handle any crash case in the utility function

Reply:
Ok. Cool.

Reply:
If that happens, we can treat it as a bug in conversion and fix it.

Reply:
Yes, by that we will not couple conversion with validation.